src/bluesky/tests/test_utils.py

```diff
@@ -499,3 +501,33 @@ def kickoff(self, *args, **kwargs):
     # Called without kwargs doesn't warn
     warn_if_msg_args_or_kwargs(msg, device.kickoff, (), {})
     assert len(recwarn) == 0
+
+
+@plan
+def sample_plan():
+    return (yield Msg("null"))
+
+
+def iterating_plan():
+    yield from sample_plan()
+
+
+def non_iterating_plan():
+    yield from sample_plan()
+    sample_plan()
+
+
+@pytest.mark.parametrize("gen_func, iterated", [
+    (non_iterating_plan, False),
+    (iterating_plan, True)
+])
+def test_warning_behavior(gen_func, iterated):
```

Comment:
Please can I see an example of the warning that is created? A slightly more detailed regex passed to `match` would be ideal...

Reply:
Sure, here's an example warning thats created, happy to add your suggestion
```
/Users/vshekar/Code/bluesky/src/bluesky/utils/__init__.py:1892: RuntimeWarning:   File "/Users/vshekar/Code/misc/run-engine-tests/bluesky_tests.py", line 20, in my_plan
    sleep(10)
RuntimeWarning: plan `sleep` was never iterated, did you mean to use `yield from`?
  warnings.warn(warning_message, RuntimeWarning, stacklevel=1)
  ```

Reply:
Can we put one more newline in please?
```
/Users/vshekar/Code/bluesky/src/bluesky/utils/__init__.py:1892: RuntimeWarning:
File "/Users/vshekar/Code/misc/run-engine-tests/bluesky_tests.py", line 20, in my_plan
    sleep(10)
RuntimeWarning: plan `sleep` was never iterated, did you mean to use `yield from`?
  warnings.warn(warning_message, RuntimeWarning, stacklevel=1)
```

Reply:
Added in https://github.com/bluesky/bluesky/pull/1709/commits/fa2131335c8cff73907d70fddc0df556b4c93eba